systemFlake.nix

```diff
@@ -48,34 +53,42 @@ let
     "checksBuilder"
   ];
 
-  nixosConfigurationBuilder = hostname: profile: (
-    # It would be nice to get `nixosSystem` reference from `selectedNixpkgs` but it is not possible at this moment
-    inputs."${channelNameFromProfile profile}".lib.nixosSystem (genericConfigurationBuilder hostname profile)
-  );
-
   getNixpkgs = profile: self.pkgs."${systemFromProfile profile}"."${channelNameFromProfile profile}";
 
-  genericConfigurationBuilder = hostname: profile: (
-    let selectedNixpkgs = getNixpkgs profile; in
+  configurationBuilder = hostname: profile: (
+    let
+      selectedNixpkgs = getNixpkgs profile;
+      defaultBuilder =
+        if defaultConfigBuilder == null
+        then inputs."${channelNameFromProfile profile}".lib.nixosSystem
+        else defaultConfigBuilder;
+      builder = profile.builder or defaultBuilder;
+    in
     {
-      inherit (selectedNixpkgs) system;
-      modules = [
-        ({ pkgs, lib, ... }: {
-          networking.hostName = hostname;
-
-          nixpkgs = {
-            inherit (selectedNixpkgs) overlays config system;
-          };
-
-          system.configurationRevision = lib.mkIf (self ? rev) self.rev;
-          nix.package = lib.mkDefault pkgs.nixUnstable;
-        })
-      ]
-      ++ sharedModules
-      ++ (profile.modules or [ ]);
-      extraArgs = { inherit inputs; } // sharedExtraArgs // profile.extraArgs or { };
+      name = profile.output or defaultConfigOutput;
+      value.${hostname} = builder {
+        inherit (selectedNixpkgs) system;
+        modules = [
+          ({ pkgs, lib, ... }: {
+            networking.hostName = hostname;
+
+            nixpkgs = {
+              inherit (selectedNixpkgs) overlays config system;
+            };
+
+            system.configurationRevision = lib.mkIf (self ? rev) self.rev;
+            nix.package = lib.mkDefault pkgs.nixUnstable;
```

Comment:
I think these work on darwin, but would deny home-manager support. I have a method to only set options if they exist, but it adds a bit of complexity:
```nix
{
type = "merge";
contents = [
  (optional (options ? nixpkgs) { nixpkgs = ... })
];
}
```

Reply:
This would make this PR **actually** config system agnostic.

Reply:
And then to ensure pkgs always gets passed at the very least it could be:
`if (options ? pkgs) { nixpkgs = ... } else { _module.args.pkgs = ... }`